diff --git a/src/main/Environment.cpp b/src/main/Environment.cpp
index bc9f852..cd7a431 100644
--- a/src/main/Environment.cpp
+++ b/src/main/Environment.cpp
@@ -65,7 +65,6 @@ const axutil_env_t* WSF_CALL Environment::getEnv()
 		}else
 		{
 			printf("Environment creation failed");
-			exit(0);
 		}
 	}		
 	return env;
@@ -120,8 +119,6 @@ void WSF_CALL Environment::initialize(std::string logFileName, axutil_log_levels
 	axutil_env_t *env = axutil_env_create_all(logFileName.c_str(), logLevel);
 	if(env){
 		setEnv(env);
-	}else{
-		exit(0);
 	}
 }
 
diff --git a/wsf_c/axis2c/util/src/dir_handler.c b/wsf_c/axis2c/util/src/dir_handler.c
index 4a5e51e..d75b2e4 100644
--- a/wsf_c/axis2c/util/src/dir_handler.c
+++ b/wsf_c/axis2c/util/src/dir_handler.c
@@ -200,7 +200,7 @@ axutil_dir_handler_list_service_or_module_dirs(
     AXIS2_ENV_CHECK(env, NULL);
     file_list = axutil_array_list_create(env, 0);
     if(!AXIS2_GETCWD(cwd, 500))
-        exit(1);
+        return NULL;
 
     /* pathname is path of services directory or modules directory. */
     chdir_result = AXIS2_CHDIR(pathname);
